### PR TITLE
[SYCL][COMPAT] Add default constructor for dim3

### PR DESCRIPTION
**Pull request by aacostadiaz**
This PR  add a default constructor for the Dim3 class.

### CODE REVIEW

**Comment by Alcpz** [comment]
Could you change the default constructor to set dimensions to `0`? I think its better to force the user to always specify their workgroup size.

**Review by Alcpz — APPROVED** [review]

**Comment by aacostadiaz** [comment]
@llvm/intel-gatekeepers This is ready to merge. Thanks.

File: sycl/include/syclcompat/dims.hpp [hunk 1]
```diff
@@ -38,6 +38,8 @@ class dim3 {
 
   dim3(const sycl::range<1> &r) : x(r[0]), y(1), z(1) {}
 
+  constexpr dim3() : x(0), y(0), z(0) {}
```
Comment (mehdi-goli) [inline sycl/include/syclcompat/dims.hpp:41]: Do you need to set this to zero. The value should not be used and I assume all you need here is to be able to instantiate this class and set the dim later on. Can you set it to   `dim3() = default;`

**Review by aacostadiaz — COMMENTED** [review]
Actually `x`, `y` & `z` are const. We can't define this constructor like that.

**Comment by Alcpz** [comment]
You're right. I've removed your comment requesting the merge. Removing `const` from the attributes would require modifying all of our constexpr constructors and would reduce a bit of performance. We haven't encountered a case where we needed to modify the original dim3 object, so if there is no alternative, feel free to include the changes to the header that would make this work. This would include modifying the documentation and maybe creating a test

**Comment by joeatodd** [comment]
Closing for now. Please re-open as/when required @aacostadiaz

**Pull request closed without merging**